src/extension.ts

```diff
@@ -59,7 +78,13 @@ export async function activateInternal(context: vscode.ExtensionContext, perfSta
         const validateEventId: string = 'azureFunctions.validateFunctionProjects';
         // tslint:disable-next-line:no-floating-promises
         callWithTelemetryAndErrorHandling(validateEventId, async (actionContext: IActionContext) => {
-            await verifyVSCodeConfigOnActivate(actionContext, vscode.workspace.workspaceFolders);
+            if (ext.context.globalState.get(GlobalStates.initProjectWithoutConfigVerification) === true) {
```

Comment:
Just call `initProjectForVSCode` before you reload the window and you shouldn't have to mess with the globalState

Reply:
verifyVSCodeConfigOnActivate is existing code for all other scenarios. Even if I change to always call initProjectForVSCode , we would still need the global state to determine whether to pass in the filePath and language so that users don't need to do the selection again for our scenario as we already know the language from the uri.

If there is any confusion here, it would be better to discuss it over a call to expedite this process.

Reply:
Not sure I quite follow this one. When I revert this code from extension.ts and call initProjectForVSCode at the end of 
setupProjectFolder try block, I am getting the below message when I run the code.

![image](https://user-images.githubusercontent.com/14221995/104257300-f24a0500-5431-11eb-96a4-50d89fe79a1c.png)


With my current code with the global states I see the following which is initializing it fine:
![image](https://user-images.githubusercontent.com/14221995/104257760-004c5580-5433-11eb-87e9-bd7b697aa623.png)

Reply:
Seems like updating to force it to open in a new window based on your other comment fixed this as well. I will update the code and remove the unneeded global states